librz/analysis/arch/arm/arm_il32.c

```diff
@@ -252,6 +406,67 @@ static RzILOpBitVector *arg_mem(RzILOpBitVector *base_plus_disp, cs_arm_op *op,
 	return base_plus_disp;
 }
 
+/**
+ * For VFP/NEON instruction immediate value
+ * <imm> in Arm ref manual: "A constant of the type specified by <dt>.
+ * This constant is replicated enough times to fill the destination register.
+ */
+static RzILOpBitVector *repeated_imm(ut32 imm_width, ut32 dreg_width, ut32 imm) {
+	ut64 final_imm = 0;
+	ut32 repeat_times = dreg_width / imm_width;
+	ut64 tmp = imm;
+
+	if (dreg_width == 128) {
+		// for <Qd> registers
+		ut64 imm_low = 0;
+		ut64 imm_high = 0;
+
+		for (int i = 0; i < repeat_times / 2; ++i) {
+			imm_low += tmp;
+			imm_high += tmp;
```

Comment:
shouldn't these be logical OR instead of ADD operations?

Reply:
it could be discussed later. I think a repeated immediate value would be much clear than immediate value with multiple OR. Though logical OR of the immediate value seems more precise from semantic perspective.